src/test/java/emissary/util/io/LoggingPrintStreamTest.java

```diff
@@ -0,0 +1,305 @@
+import ch.qos.logback.core.ConsoleAppender;
+import ch.qos.logback.core.read.ListAppender;
+import org.apache.commons.io.output.NullOutputStream;
+import org.apache.commons.lang3.Validate;
+import org.apache.log4j.MDC;
+import org.junit.jupiter.api.Test;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static emissary.util.io.LoggingPrintStream.NORMAL_SEPARATOR;
+import static emissary.util.io.LoggingPrintStream.THROWABLE_PREFIX;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class LoggingPrintStreamTest {
+    private static final String LOG_MSG_1 = "Log message NUMBER 1";
+    private static final String LOG_MSG_2 = "Log message NUMBER 2";
+    private static final Object LOG_OBJ_1 = Integer.valueOf(1234567890);
+    private static final Exception EXCEPTION_CAUSE = new Exception("Test Exception 'cause'");
+    private static final Exception EXCEPTION_ONE = new Exception("Test Exception NUMBER 1", EXCEPTION_CAUSE);
+    private static final Exception EXCEPTION_TWO = new Exception("Test Exception NUMBER 2", EXCEPTION_CAUSE);
+
+    @Test
+    void testArguments() {
+        final Logger logger = (Logger) LoggerFactory.getLogger(LoggingPrintStream.class);
+        final org.slf4j.event.Level slf4jLevel = org.slf4j.event.Level.TRACE;
+
+        assertThrows(NullPointerException.class, () -> new LoggingPrintStream(null, "TEST", logger, slf4jLevel, 1, TimeUnit.SECONDS));
+        assertThrows(NullPointerException.class, () -> new LoggingPrintStream(System.out, null, logger, slf4jLevel, 1, TimeUnit.SECONDS));
+        assertThrows(NullPointerException.class, () -> new LoggingPrintStream(System.out, "TEST", null, slf4jLevel, 1, TimeUnit.SECONDS));
+        assertThrows(NullPointerException.class, () -> new LoggingPrintStream(System.out, "TEST", logger, null, 1, TimeUnit.SECONDS));
+        assertThrows(IllegalArgumentException.class, () -> new LoggingPrintStream(System.out, "TEST", logger, slf4jLevel, -1, TimeUnit.SECONDS));
+        assertThrows(NullPointerException.class, () -> new LoggingPrintStream(System.out, "TEST", logger, slf4jLevel, 1, null));
+    }
+
+    @Test
+    void testClose() throws Exception {
+        final Logger logger = (Logger) LoggerFactory.getLogger(LoggingPrintStream.class);
+        final org.slf4j.event.Level slf4jLevel = org.slf4j.event.Level.TRACE;
+
+        try (LoggingPrintStream loggingPrintStream = new LoggingPrintStream(System.out, "TEST", logger, slf4jLevel, 30, TimeUnit.SECONDS)) {
```

Comment:
When I run test this in IntelliJ, this test exits with a "terminated" status, and any remaining tests in this class then get the same status.  I believe this is happening because the LoggingPrintStream.close() method is closing `System.out` which I think is messing with the IDE's/JUnit's intercepted stream.   

```suggestion
        try (LoggingPrintStream loggingPrintStream = new LoggingPrintStream(new ByteArrayOutputStream(), "TEST", logger, slf4jLevel, 30, TimeUnit.SECONDS)) {
```